lib/rosette/integrations/smartling-integration/smartling_pusher.rb

```diff
@@ -4,101 +4,64 @@ module Rosette
   module Integrations
     class SmartlingIntegration < Integration
       class SmartlingPusher
-        FILE_TYPES =
-          %w(android ios gettext html javaProperties yaml xliff xml json docx pptx xlsx idml qt resx plaintext)
 
-        SERIALIZER_FILE_TYPE_MAP = {
-          'xml/android' => 'android'
-        }
+        attr_reader :rosette_config, :repo_config
 
-        attr_reader :rosette_config, :integration_config, :repo_name, :smartling_api
-
-        def initialize(rosette_config, integration_config, repo_name, smartling_api)
-          @smartling_api = smartling_api
-          @integration_config = integration_config
+        def initialize(rosette_config)
           @rosette_config = rosette_config
-          @repo_name = repo_name
+        end
+
+        def set_repo_config(repo_config)
+          @repo_config = repo_config
+          self
         end
 
         def push(commit_id, serializer_id)
           phrases = phrases_for(commit_id)
 
-          serializer_const = Rosette::Core::SerializerId.resolve(serializer_id)
-          dest_filename = destination_filename_for(commit_id, serializer_const)
-
-          file_for_upload(phrases, serializer_const) do |tmp_file|
-            response = smartling_api.upload(
-              tmp_file.path,
-              dest_filename,
-              file_type_for(serializer_id), {
-                approved: smartling_api.preapprove_translations?
-              }
+          if phrases.size > 0
+            file_name = file_name_for(commit_id)
+            uploader = build_uploader_for(
+              phrases, file_name, serializer_id
             )
 
+            response = uploader.upload
             phrase_count = response['stringCount']
+
             rosette_config.datastore.add_or_update_commit_log(
-              repo_name, commit_id, nil, Rosette::DataStores::PhraseStatus::PENDING, phrase_count
+              repo_config.name, commit_id, nil,
+              Rosette::DataStores::PhraseStatus::PENDING, phrase_count
             )
           end
         rescue => ex
-          Rosette.logger.error('Caught an exception while pushing to Smartling API.')
-          Rosette.logger.error("#{ex.message}\n#{ex.backtrace.join("\n")}")
-          raise ex
+          rosette_config.error_reporter.report_error(ex)
```

Comment:
you may be able to use a common method to get at the logger across the various classes.